### DISCUSSION TITLE
InteractiveWebAssembly vs InteractiveAuto

### DISCUSSION THREAD

**Discussion by 0qie** [dotnet/aspnetcore#52553] [Q&A]
I don’t think there is any difference in the current state. In Tim Corey’s video, you can see that even when InteractiveWebAssembly is specified, it still shows from server side and then switches to webassembly. But that’s what InteractiveAuto should do, not InteractiveWebAssembly. Here is the link to the video: [Tim Corey’s video](https://www.youtube.com/watch?v=walv3nLTJ5g&t=2220s)

Maybe I am missing something important?

**--UPDATE--**

I Searched on Bing (yes, that’s right. No hard feelings, Google) and found a post by @danroth27 that said this:
![image](https://github.com/dotnet/aspnetcore/assets/22215187/6b507f80-2684-46d7-9900-cc68e6e73c06)

This means that prerendering is turned on by default, which I don’t think is a good idea. Shouldn’t it be off by default instead? Otherwise, how is it different?

**Comment by KennethHoff** [comment]
Prerending is the act of generating some HTML on the server alongside the logic (non-interactive)

WebAssembly is using WebAssembly to add interactivity to the page.

Auto chooses the best interactive mode ( server or WebAssembly) based on some criteria. Prerending is separate from that process

**Comment by MmMapIoSpace** [comment]
Can't we make the main page stay in Blazor Server Interactive, but specifically for the chat page that requires SignalR authentication in WebAssembly ?
